packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -730,53 +727,54 @@ function setupMessageChannel(channel: MessageChannel): ThunkAction<void, State,
   }
 }
 
-export type Store = ReduxStore<State, Action> & { dispatch: Dispatch } & SetupTeardownMixin
-
 export function configureStore({
   preloadedState,
   client,
 }: {
   preloadedState: Partial<State>
   client: MakeswiftHostApiClient
-}): Store {
+}) {
   const initialState: Partial<State> = {
     ...preloadedState,
     isPreview: IsPreview.getInitialState(true),
   }
 
-  const composeEnhancers = composeWithDevToolsDevelopmentOnly({
-    name: `Host store (${new Date().toISOString()})`,
-    serialize: true,
-    stateSanitizer: (state: any) => serializeState(state),
-    actionsDenylist: [
-      ActionTypes.BUILDER_POINTER_MOVE,
-      ActionTypes.HANDLE_POINTER_MOVE,
-      ActionTypes.ELEMENT_FROM_POINT_CHANGE,
-    ],
-  })
-
   const channel = new MessageChannel()
-  const store = createStore(
+  const store = configureReduxStore({
     reducer,
-    initialState,
-    composeEnhancers(
-      withSetupTeardown(
-        () => {
-          const dispatch = store.dispatch as Dispatch
-          dispatch(setupMessageChannel(channel))
-        },
-        () => channel.teardown(),
-      ),
-      applyMiddleware(
-        thunk,
+    preloadedState: initialState,
+
+    middleware: getDefaultMiddleware =>
+      getDefaultMiddleware(middlewareOptions).concat(
         ReactPage.elementTreeMiddleware(),
         measureBoxModelsMiddleware(),
         messageChannelMiddleware(channel),
         propControllerHandlesMiddleware(),
         makeswiftApiClientSyncMiddleware(client),
       ),
```

Comment:
Same list of middleware as before except for `thunk`, which is included by default.